docs/features/search/search-engines.md

```diff
@@ -0,0 +1,138 @@
+---
+id: search-engines
+title: Search Engines
+description: Choosing and configuring your search engine for Backstage
+---
+
+# Search Engines
+
+Backstage supports 2 search engines by default, an in-memory engine called Lunr
+and ElasticSearch. You can configure your own search engines by implementing the
+provided interface as mentioned in the
+[search backend documentation.](./getting-started.md#Backend)
+
+Provided search engine implementations have their own way of constructing
+queries, which may be something you want to modify. Alterations to the querying
+logic of a search engine can be made by providing your own implementation of a
+QueryTranslator interface. This modification can be done without touching
+provided search engines by using the exposed setter to set the modified query
+translator into the instance.
+
+```
+const searchEngine = new LunrSearchEngine({ logger });
+searchEngine.setTranslator(new MyNewAndBetterQueryTranslator());
+```
+
+## Lunr
+
+Lunr search engine is enabled by default for your backstage instance if you have
+not done additional changes to the scaffolded app.
+
+Lunr can be instantiated like this:
+
+```typescript
+// app/backend/src/plugins/search.ts
+const searchEngine = new LunrSearchEngine({ logger });
+const indexBuilder = new IndexBuilder({ logger, searchEngine });
+```
+
+## ElasticSearch
+
+Backstage supports ElasticSearch search engine connections, indexing and
+querying out of the box. Available configuration options enable usage of either
+AWS or Elastic.co hosted solutions, or a custom self-hosted solution.
+
+Similarly to Lunr above, ElasticSearch can be set up like this:
+
+```typescript
+// app/backend/src/plugins/search.ts
```

Comment:
We should mention that one also needs to add the `@backstage/plugin-search-backend-module-elasticsearch` to the backend package.